src/app/state/account/saga.test.ts

```diff
@@ -0,0 +1,15 @@
+import { testSaga } from 'redux-saga-test-plan'
+import { accountActions } from '.'
+import { accountSaga, loadAccount, refreshAccountOnTransaction } from './saga'
+
+describe('Account Sagas', () => {
+  test('accountSaga', () => {
+    testSaga(accountSaga)
+      .next()
+      .fork(refreshAccountOnTransaction)
+      .next()
+      .takeLatest(accountActions.fetchAccount, loadAccount)
+      .next()
+      .isDone()
```

Comment:
I don't think this is a useful test.

I'd rather see approximately
```ts
expectSaga(stakingSaga)
  .provide(getAccount, () => delay(10*Math.random()))
  .put(fetchAccount('a'))
  .put(fetchAccount('b'))
  .hasFinalState({ address: 'b' })
  .run()
```

Reply:
I think this unit test if valid unless I am missing something, but I am following one of redux saga test plan pattern. I am adding a unit test not integration test. Unit test is testing a watcher saga only and it doesn't care about integration with other parts of code. The bug exists because of using a wrong effect creator. Unit test is validating if a correct effect creator is used so it covers current bug/potential regression bug.